vnext/Microsoft.ReactNative/IReactDispatcher.h

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+#include "ReactDispatcherHelper.g.h"
+#include <ReactPropertyBag.h>
+#include <dispatchQueue/dispatchQueue.h>
+#include <winrt/Microsoft.ReactNative.h>
+
+namespace winrt::Microsoft::ReactNative {
+
+struct ReactDispatcher : implements<ReactDispatcher, IReactDispatcher> {
+  ReactDispatcher() = default;
+  ReactDispatcher(Mso::DispatchQueue &&queue) noexcept;
+
+  bool HasThreadAccess() noexcept;
+  void Post(ReactDispatcherCallback const &callback) noexcept;
```

Comment:
>& [](start = 42, length = 1)

Why not &&?

Reply:
This is C++/WinRT thing : they pass parameters by const&. I will check again if we can use && in places like this, but my previous tries were not successful.

---
In reply to: [423967144](https://github.com/microsoft/react-native-windows/pull/4877#discussion_r423967144) [](ancestors = 423967144)

Reply:
I tried it. It does not compile. The C++/WinRT uses const&.

---
In reply to: [423983290](https://github.com/microsoft/react-native-windows/pull/4877#discussion_r423983290) [](ancestors = 423983290,423967144)